corehq/apps/userreports/reports/specs.py

```diff
@@ -393,6 +394,28 @@ def get_query_column_ids(self):
         return [self.column_id]
 
 
+class SumWhenColumn(ConditionalAggregationColumn):
+    type = TypeProperty("sum_when")
+    else_ = IntegerProperty()
+
+    def get_column_config(self, data_source_config, lang):
+        return ColumnConfig(columns=[
+            DatabaseColumn(
+                header=self.get_header(lang),
+                agg_column=SumWhen(
+                    whens=self.whens,
+                    else_=self.else_,
+                    alias=self.column_id,
+                ),
+                sortable=self.sortable,
+                data_slug=self.column_id,
+                format_fn=self.get_format_fn(),
+                help_text=self.description,
+                visible=self.visible,
+            )],
+        )
```

Comment:
This could also define `get_whens` as `return self.whens` and not override `get_column_config`.  If we end up with any more of these, it'd be nice to share more code and make the differences obvious.